services/management/commands/verify_search_index_integrity.py

```diff
@@ -0,0 +1,59 @@
+from django.conf import settings
+from django.core.exceptions import FieldError
+from django.core.management.base import BaseCommand, CommandError
+from django.utils import translation
+
+from haystack.query import SearchQuerySet
+
+from services.models import Unit, Service
+from services.search_indexes import AdministrativeDivisionIndex
+from munigeo.models import Address, AdministrativeDivision
+
+
+LANGUAGES = [l[0] for l in settings.LANGUAGES]
+
+
+def _check_index_integrity(model):
+    errors = []
+    # Unit._meta.label_lower
+
+    def format_error(model=None, public=None, db=None, index=None, language=None):
+        return ("Differing count for language {language} model {model} "
+                "with public=={public}, db: {db}, index: {index}").format(
+                    model=model, public=public, db=db, index=index, language=language)
+
+    for language in LANGUAGES:
+        with translation.override(language):
+            for public in [True, False]:
```

Comment:
Why is `public` string (`'true'`) in one place and boolean (`True`) here?

Reply:
Well, the django orm filters handle boolean type filters okay, so they need the values as boolean. However, haystack seems to have a problem with boolean fields at least with elasticsearch. The filter simply only works with string values.

Reply:
What's weird is that the field is configured as a boolean field in elastic, but still haystack silently ignores boolean values in filters, probably because it simply converts them to the strings "True" and "False" (uppercase).